core/src/main/java/nl/nn/adapterframework/core/IConfigurationAware.java

```diff
@@ -0,0 +1,25 @@
+/*
+   Copyright 2021 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.core;
+
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+public interface IConfigurationAware extends IScopeProvider, ApplicationContextAware {
+
+	public String getName();
+	public ApplicationContext getApplicationContext();
+}
```

Comment:
Door deze constructie krijgen nu heel veel classes zowel een applicationContext als een configurationClassLoader.
Zouden we niet kunnen volstaan met een applicationContext? Dat die ook gebruikt wordt om scope te providen?

Reply:
Ja dat wil ik in een volgende pr doen, anders wordt het echt uit de hand!